src/test/java/io/airlift/units/TestDataSizeValidator.java

```diff
@@ -79,57 +79,48 @@ public void testAllowsNullMaxAnnotation()
     @Test
     public void testDetectsBrokenMinAnnotation()
     {
-        try {
-            VALIDATOR.validate(new BrokenMinAnnotation());
-            fail("expected a ValidationException caused by an IllegalArgumentException");
-        }
-        catch (ValidationException e) {
-            assertThat(e).hasRootCauseInstanceOf(IllegalArgumentException.class);
-        }
+        assertThatThrownBy(() -> VALIDATOR.validate(new BrokenMinAnnotation()))
+                .isInstanceOf(ValidationException.class)
+                .hasRootCauseInstanceOf(IllegalArgumentException.class)
+                .hasMessage("java.lang.IllegalArgumentException: size is not a valid data size string: broken");
     }
 
     @Test
     public void testDetectsBrokenMaxAnnotation()
     {
-        try {
-            VALIDATOR.validate(new BrokenMaxAnnotation());
-            fail("expected a ValidationException caused by an IllegalArgumentException");
-        }
-        catch (ValidationException e) {
-            assertThat(e).hasRootCauseInstanceOf(IllegalArgumentException.class);
-        }
+        assertThatThrownBy(() -> VALIDATOR.validate(new BrokenMaxAnnotation()))
+                .isInstanceOf(ValidationException.class)
+                .hasRootCauseInstanceOf(IllegalArgumentException.class)
+                .hasMessage("java.lang.IllegalArgumentException: size is not a valid data size string: broken");
     }
 
     @Test
     public void testPassesValidation()
     {
-        ConstrainedDataSize object = new ConstrainedDataSize(new DataSize(7, MEGABYTE));
-        Set<ConstraintViolation<ConstrainedDataSize>> violations = VALIDATOR.validate(object);
-        assertTrue(violations.isEmpty());
+        assertTrue(VALIDATOR.validate(new ConstrainedDataSize(new DataSize(7, MEGABYTE)))
+                .isEmpty());
     }
 
     @Test
     public void testFailsMaxDataSizeConstraint()
     {
-        ConstrainedDataSize object = new ConstrainedDataSize(new DataSize(11, MEGABYTE));
-        Set<ConstraintViolation<ConstrainedDataSize>> violations = VALIDATOR.validate(object);
+        Set<? extends ConstraintViolation<?>> violations = VALIDATOR.validate(new ConstrainedDataSize(new DataSize(11, MEGABYTE)));
         assertThat(violations).hasSize(2);
-
-        for (ConstraintViolation<ConstrainedDataSize> violation : violations) {
-            assertThat(violation.getConstraintDescriptor().getAnnotation()).isInstanceOf(MaxDataSize.class);
-        }
+        assertThat(violations.stream().map(violation -> violation.getConstraintDescriptor().getAnnotation()))
```

Comment:
FYI, there is an `extracting` method, though being able to use a stream is nice (I didn't know that worked). This lets us do
```java
assertThat(violations)
        .extracting(violation -> violation.getConstraintDescriptor().getAnnotation())
        .allSatisfy(annotation -> assertThat(annotation).isInstanceOf(MaxDataSize.class));
```
With the benefit that you get a better error message on failure

Reply:
When the final ` .containsOnly(` fails, the exception message is the same. 
But I like the `.extracting`!